amy/fiscal/tests/test_filters.py

```diff
@@ -0,0 +1,454 @@
+                "nonpositive_remaining_seats_only",
+                "order_by",
+            },
+        )
+
+    def test_filter_active_only(self):
+        # Arrange
+        name = "active_only"
+        value = True
+
+        # Act
+        result = self.filterset.filters[name].filter(self.qs, value)
+
+        # Assert
+        self.assertTrue(self.membership in result)
+        self.assertFalse(self.membership2 in result)
+
+    def test_filter_organization_name(self):
+        # Arrange
+        filter_name = "organization_name"
+        value = "Test Organization"
+
+        # Act
+        result = self.filterset.filters[filter_name].filter(self.qs, value)
+
+        # Assert
+        self.assertQuerysetEqual(result, [self.membership])
+
+    def test_filter_variant(self):
+        # Arrange
+        filter_name = "variant"
+        value = "Bronze"
+
+        # Act
+        result = self.filterset.filters[filter_name].filter(self.qs, value)
+
+        # Assert
+        self.assertTrue(self.membership in result)
+        self.assertFalse(self.membership2 in result)
+
+    def test_filter_contribution_type(self):
+        # Arrange
+        filter_name = "contribution_type"
+        value = "Financial"
+
+        # Act
+        result = self.filterset.filters[filter_name].filter(self.qs, value)
+
+        # Assert
+        self.assertTrue(self.membership in result)
+        self.assertFalse(self.membership2 in result)
+
+    def test_filter_training_seats_only(self):
+        # Arrange
+        filter_name = "training_seats_only"
+        value = True
+
+        # Act
+        result = self.filterset.filters[filter_name].filter(self.qs, value)
+
+        # Assert
+        self.assertTrue(self.membership in result)
+        self.assertFalse(self.membership2 in result)
+
+    def test_filter_nonpositive_remaining_seats_only(self):
+        # Arrange
+        filter_name = "nonpositive_remaining_seats_only"
+        value = True
+
+        # Act
+        result = self.filterset.filters[filter_name].filter(self.qs, value)
+
+        # Assert
+        self.assertTrue(self.membership in result)
+        self.assertFalse(self.membership2 in result)
+
+    def test_filter_order_by(self):
+        # Arrange
+        filter_name = "order_by"
+        fields = self.filterset.filters[filter_name].param_map
+        results = {}
+        # default ordering is ascending
+        expected_results = {
+            "agreement_start": [self.membership2, self.membership],
+            "agreement_end": [self.membership2, self.membership],
+            "instructor_training_seats_remaining": [self.membership, self.membership2],
+        }
+
+        # Act
+        for field in fields.keys():
+            results[field] = self.filterset.filters[filter_name].filter(
+                self.qs, [field]
+            )
+
+        # Assert
+        # we don't have any unexpected fields
+        self.assertListEqual(list(fields.keys()), list(expected_results.keys()))
```

Comment:
Does `assertEqual(fields.keys(), expected_results.keys())` also work? `.keys()` returns a set-like object, so we should be able to compare them.

Reply:
It does, I think I iterated this a few times and then didn't think about simplifying 😄